Sources/Workspace/Workspace.swift

```diff
@@ -1262,16 +1266,25 @@ extension Workspace {
         for (specifier, state) in packageStateChanges {
             diagnostics.wrap {
                 switch state {
-                case .added(let requirement):
-                    _ = try clone(specifier: specifier, requirement: requirement)
-                case .updated(let requirement):
-                    _ = try clone(specifier: specifier, requirement: requirement)
-                case .removed:
-                    try remove(specifier: specifier)
-                case .unchanged: break
+                    case .added(let requirement):
+                        _ = try clone(specifier: specifier, requirement: requirement)
+                    case .updated(let requirement):
+                        _ = try clone(specifier: specifier, requirement: requirement)
+                    case .removed:
+                        try remove(specifier: specifier)
+                    case .unchanged:
+                        break
                 }
             }
         }
+        
+        // Get the count of `.unchanged` package state changes.
+        let unchangedCount = packageStateChanges.filter({ if case .unchanged = $0.value { return true }; return false }).count
+        
+        if unchangedCount == packageStateChanges.count
+        {
+            delegate.repositoriesRemainedUnchanged(packageStateChanges.keys.map({ $0.url }))
```

Comment:
I think the delegate is kind of misleading in a way that one can expect it to be called with the list of repositories that were not updated. Lets just change it to something simple like `delegate.dependenciesUpToDate()` and remove the url array.